src/rust/engine/protos/protos/pants/cache.proto

```diff
@@ -15,6 +15,7 @@ message CacheKey {
   CacheKeyType key_type = 1;
 
   build.bazel.remote.execution.v2.Digest digest = 2;
+  optional string metadata = 3;  // Any other metadata to accompany the tag, e.g the JSON request of a DEP_INFERENCE_REQUEST.
 }
```

Comment:
So as to https://github.com/pantsbuild/pants/issues/18961 we might include a `version` string here as well. But we can do that separately.